nms/app/packages/magmalte/app/views/equipment/FEGGatewayTable.js

```diff
@@ -0,0 +1,231 @@
+ * each federation gateway.
+ */
+export default function GatewayTable() {
+  const ctx = useContext(FEGGatewayContext);
+  const [refresh, setRefresh] = useState(true);
+
+  return (
+    <>
+      <CardTitleRow
+        key="title"
+        icon={CellWifiIcon}
+        label={`Federation Gateways (${Object.keys(ctx.state).length})`}
+        filter={() => (
+          <Grid container justify="flex-end" alignItems="center" spacing={1}>
+            <Grid item>
+              <AutorefreshCheckbox
+                autorefreshEnabled={refresh}
+                onToggle={() => setRefresh(current => !current)}
+              />
+            </Grid>
+          </Grid>
+        )}
+      />
+      <StatusTable refresh={refresh} />
+    </>
+  );
+}
+
+/**
+ * Returns a table containing the federation gateways which shows their
+ * basic configurations.
+ * @param {boolean} refresh Tells to autorefresh after 30 seconds or not.
+ */
+function GatewayStatusTable(props: WithAlert & {refresh: boolean}) {
+  const {history, relativeUrl, match} = useRouter();
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const networkId: string = nullthrows(match.params.networkId);
+  const gwCtx = useContext(FEGGatewayContext);
+  const [lastRefreshTime, setLastRefreshTime] = useState(
+    new Date().toLocaleString(),
+  );
+  // Auto refresh gateways every 30 seconds
+  const state = useRefreshingContext({
+    context: FEGGatewayContext,
+    networkId: networkId,
+    type: RefreshTypeEnum.FEG_GATEWAY,
+    interval: REFRESH_INTERVAL,
+    enqueueSnackbar,
+    refresh: props.refresh,
+    lastRefreshTime: lastRefreshTime,
+  });
+  const fegGateways = state?.fegGateways || {};
+  const health = state?.health || {};
+  const activeFegGatewayId = state?.activeFegGatewayId || '';
+  const ctxValues = [...Object.values(gwCtx.state)];
+  useEffect(() => {
+    setLastRefreshTime(new Date().toLocaleString());
+  }, [ctxValues.length]);
+
+  const [currRow, setCurrRow] = useState<EquipmentFegGatewayRowType>({});
+  const fegGatewayRows: Array<EquipmentFegGatewayRowType> = [];
+  Object?.keys(fegGateways)
+    .map((gwId: string) => fegGateways[gwId])
+    .filter((g: federation_gateway) => g.federation && g.id)
+    .map((gateway: federation_gateway) => {
+      fegGatewayRows.push({
+        name: gateway.name,
+        id: gateway.id,
+        is_primary: activeFegGatewayId === gateway.id,
+        health: health[gateway.id].status
+          ? health[gateway.id]?.status === HEALTHY_STATUS
+            ? HEALTHY_GATEWAY
+            : UNHEALTHY_GATEWAY
+          : '-',
+        gx: gateway.federation?.gx?.server?.address || '-',
+        gy: gateway.federation?.gy?.server?.address || '-',
+        swx: gateway.federation?.swx?.server?.address || '-',
+        s6a: gateway.federation?.s6a?.server?.address || '-',
+        csfb: gateway.federation?.csfb?.client?.server_address || '-',
+      });
+    });
+
+  return (
```

Comment:
We should probably display something different if there's no federation gateways. Say, if they just haven't been configured yet, the NMS would just appear broken. Defer to @jingtien if necessary.

Reply:
I don't understand why it would appear broken. I am saying this because if there are no federation gateways in the network then all the headers of the table are there(name, id, gx ,...), but there is no data in there. Anyone seeing that would notice that there is no federation gateway to display under those headers. If you created a new test lte network go to the equipment page without adding any gateway, a blank table will be seen there. 
<img width="1697" alt="Screen Shot 2021-07-21 at 2 33 13 PM" src="https://user-images.githubusercontent.com/34602743/126541339-4a169cfb-b48e-4590-a9ba-c2127076b1e9.png">

I think anyone seeing this would notice that there are no gateways in the network(the same applies here too). Besides the count of 0 on table title will tell you that there are no federation gateways. Does that make sense?

Reply:
Sounds good. Thanks for the detailed response.